### PR TITLE
imlib: Add hmirror, vflip, and transpose support to draw_image.

### PR DESCRIPTION
**Pull request by kwagyeman**
For: https://github.com/openmv/openmv/issues/1963

Seems to work great. I notice it being 2x faster than the previous transpose. I hit max FPS with it on my OpenMV Cam Pure Thermal when rotating an RGB565 image. It should meet our needs to allow video playback on the Arduino Giga.

### CODE REVIEW

**Comment by iabdalkader** [comment]
This pull request has been mentioned on **OpenMV Forums**. There might be relevant details there:

https://forums.openmv.io/t/image-not-rendering-correctly-on-arduino-giga-display/8852/31

**Comment by kwagyeman** [comment]
Has to be refactored to match #1992.

**Comment by kwagyeman** [comment]
Ready for review. Dependent on #1992.

File: src/omv/imlib/imlib.h [hunk 1]
```diff
@@ -1077,10 +1077,13 @@ typedef struct find_barcodes_list_lnk_data {
...
-    IMAGE_HINT_BILINEAR = 1 << 1,
-    IMAGE_HINT_BICUBIC  = 1 << 2,
-    IMAGE_HINT_CENTER   = 1 << 7,
+    IMAGE_HINT_AREA      = 1 << 0,
+    IMAGE_HINT_BILINEAR  = 1 << 1,
+    IMAGE_HINT_BICUBIC   = 1 << 2,
+    IMAGE_HINT_HMIRROR   = 1 << 4,
+    IMAGE_HINT_VFLIP     = 1 << 5,
+    IMAGE_HINT_TRANSPOSE = 1 << 6,
+    IMAGE_HINT_CENTER    = 1 << 7,
```
Comment (iabdalkader) [inline src/omv/imlib/imlib.h:1086]: Can you wrap those with `()` ?

Comment (kwagyeman) [reply src/omv/imlib/imlib.h]: Done

**Review by iabdalkader — APPROVED** [review]

**Pull request merged**